astropy/time/core.py

```diff
@@ -1653,7 +1656,10 @@ def _time_difference(self, other, op=None):
         return (self.jd1 - other.jd1) + (self.jd2 - other.jd2)
 
     def __lt__(self, other):
-        return self._time_difference(other, '<') < 0.
+        try:
```

Comment:
It is indeed better to return `NotImplemented`, though it should be done in `_time_difference` - I'll make a quick PR that this can be rebased on.